core/src/main/java/org/apache/calcite/adapter/jdbc/JdbcRules.java

```diff
@@ -270,7 +270,7 @@ public JdbcJoinRule(JdbcConvention out) {
     public JdbcJoinRule(JdbcConvention out,
         RelBuilderFactory relBuilderFactory) {
       super(Join.class, (Predicate<RelNode>) r -> true, Convention.NONE,
-          out, relBuilderFactory, "JdbcJoinRule");
+          out, relBuilderFactory, "JdbcJoinRule:" + out);
```

Comment:
I have the impression that JdbcConvention#toString() may return a String that does not obey the description pattern of the rule and thus get a RuntimeException when instantiating the rule. Is it possible? Should we normalize somehow the string obtained from the convention?